SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessCallable.java

```diff
@@ -39,4 +61,31 @@ public String toString() {
     return "SingularityExecutorTaskProcessCallable [task=" + task + "]";
   }
 
+  private void runHealthcheck() {
+    Optional<HealthcheckOptions> maybeOptions = task.getTaskDefinition().getHealthcheckOptions();
+    Optional<String> expectedHealthcheckResultFilePath = task.getTaskDefinition().getHealthcheckResultFilePath();
+
+    if (maybeOptions.isPresent() && expectedHealthcheckResultFilePath.isPresent()) {
+      LOG.debug("Checking for healthcheck file {}", expectedHealthcheckResultFilePath.get());
+      String taskAppDirectory = task.getTaskDefinition().getTaskAppDirectory();
+      File fullHealthcheckPath = Paths.get(taskAppDirectory, expectedHealthcheckResultFilePath.get()).toFile();
+
+      try {
+        Integer healthcheckMaxRetries = maybeOptions.get().getMaxRetries().or(configuration.getDefaultHealthcheckMaxRetries());
+
+        Retryer<Boolean> retryer = RetryerBuilder.<Boolean>newBuilder()
+            .retryIfResult(bool -> !bool)
+            .withWaitStrategy(WaitStrategies.fixedWait(1L, TimeUnit.SECONDS))
```

Comment:
missed this one earlier. There is a healthcheck interval in the healthcheck options as well. Right now we are essentially treating the max retries as a timeout in seconds, which might be . bit confusing to the user given how the web health checks work. If we take the interval into account here then the math for total timeout works the same as it does for web service checks